src/movies.js

```diff
@@ -1,25 +1,139 @@
 // Iteration 1: All directors? - Get the array of all directors.
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors.
 // How could you "clean" a bit this array and make it unified (without duplicates)?
-function getAllDirectors() {}
+
+function getAllDirectors(arr) {
+  let directors = arr.map(function(el) {
+    return el.director;
+  })
+  return directors;
+  // we need to return this or it just gets undefined
+}
+
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
-function howManyMovies() {}
+
+function howManyMovies(arr) {
+
+const filtered = movies.filter(function(movie){
+  if (movie.director === 'Steven Spilberg' && movie.genre.icludes ('Drama')){
+    return filtered
+  }
+})
+
+// filter out
+// check out the range of the filtered results
 
 // Iteration 3: All scores average - Get the average of all scores with 2 decimals
-function scoresAverage() {}
+
+
+function scoresAverage(array) {
+
+const total = 0; 
+const count = 0; 
+
+array.forEach(function(item, index){
+total += item; 
+count++; 
+
+});
+
+return total/count; ht
+
+}
+
+
+// diferent way of doing it ES6 
+
+
+console.log(scoresAverage(array)); 
+
+
+function scoresAverage(array) {
+
+  const average = array.reduce((total,next) => total + array.average, 0) / arr.score.length
+}
+
+
+//
+
+
+
+
 
 // Iteration 4: Drama movies - Get the average of Drama Movies
-function dramaMoviesScore() {}
+
+function dramaMoviesScore(array) {
+
+const dramas = array.filter(function (genre) { return array.genre === 'Drama'}); 
+const average = dramas.reduce((total,next) => total + next.score, 0) / dramas.length; 
+
+console.log(average); 
+
+}
 
 // Iteration 5: Ordering by year - Order by year, ascending (in growing order)
-function orderByYear() {}
+
+function orderByYear(array) {
```

Comment:
here again, the argument you're passing is called `array` but you are iterating over `movies`